lib/fog/openstack/core.rb

```diff
@@ -119,6 +116,56 @@ def set_api_path
         # if the service supports multiple versions, do the selection here
       end
 
+      def set_microversion
+        # both need to be set in service's initialize for microversions to work
+        return if !@supported_microversion || !@supported_versions
```

Comment:
What about doing that test before, at the calling time (Line 91)? 
Because the job of set_microversion is not to test if it needs to be done, I think it's a decoupling thing.